platform/mbed_stats.c

```diff
@@ -65,6 +65,36 @@ size_t mbed_stats_stack_get_each(mbed_stats_stack_t *stats, size_t count)
     return i;
 }
 
-#if MBED_STACK_STATS_ENABLED && !MBED_CONF_RTOS_PRESENT
-#warning Stack statistics are currently not supported without the rtos.
+size_t mbed_stats_thread_get_each(mbed_stats_thread_t *stats, size_t count)
+{
+    MBED_ASSERT(stats != NULL);
```

Comment:
Should we also assert for if count == 0?

Reply:
In case of count 0, we will not fill up anything. But in case of NULL pointer, next operation in the code is `memset` which should not be performed on NULL pointer.